charts/gha-runner-scale-set-controller/templates/manager_cluster_role.yaml

```diff
@@ -126,6 +126,9 @@ rules:
   verbs:
   - list
   - watch
+  - get
+  - delete
+  - patch
```

Comment:
why do we need these cluster-wide permission?

Reply:
The role binding is the last one to get deleted
Once we get to that step, we don't have enough permissions to remove it (We removed Role in the step before)
Same problem would be if we delete first the role binding, then we won't be able to delete a role

It seemed more secure to be able to get/patch/delete role bindings than roles, so I picked that one

Reply:
Sadly no... I tried but since the finalizer is part of the `ObjectMeta`, we need patch on the actual resource in order to do this operation

Reply:
Should we flip the order to remove finalizer on the rolebinding first and then the role?
also, we might only need the `patch` permission on `role` then, I think?